src/Confluent.Kafka/Timestamp.cs

```diff
@@ -54,9 +76,14 @@ public Timestamp(DateTime dateTime, TimestampType type)
         public TimestampType Type { get; }
 
         /// <summary>
-        ///     Gets the timestamp value.
+        ///     Get the Unix millisecond timestamp.
+        /// </summary>
+        public long UnixTimestampMs { get; }
+
+        /// <summary>
+        ///     Gets the Utc DateTime corresponding to the <see cref="UnixTimestampMs"/>.
         /// </summary>
-        public DateTime DateTime { get; }
+        public DateTime DateTime => UnixTimestampMsToDateTime(UnixTimestampMs);
```

Comment:
i've been doing it for methods without braces.

Haven't looked into details of C# 7, but we're happy to require the latest released tools for development. We'd will need very good reason to stop compatibility with netstandard1.3 or net45 for usage of the library/nuget package however.